recipes/bullet3/all/conanfile.py

```diff
@@ -0,0 +1,120 @@
+        cmake.definitions["INSTALL_LIBS"] = True
+        cmake.definitions["USE_GRAPHICAL_BENCHMARK"] = self.options.graphical_benchmark
+        cmake.definitions["USE_DOUBLE_PRECISION"] = self.options.double_precision
+        cmake.definitions["BULLET2_USE_THREAD_LOCKS"] = self.options.bt2_thread_locks
+        cmake.definitions["USE_SOFT_BODY_MULTI_BODY_DYNAMICS_WORLD"] = self.options.btSoftMultiBodyDynamicsWorld
+        cmake.definitions["BUILD_ENET"] = self.options.network_support
+        cmake.definitions["BUILD_CLSOCKET"] = self.options.network_support
+        cmake.definitions["BUILD_CPU_DEMOS"] = False
+        cmake.definitions["BUILD_OPENGL3_DEMOS"] = False
+        cmake.definitions["BUILD_BULLET2_DEMOS"] = False
+        cmake.definitions["BUILD_EXTRAS"] = False
+        cmake.definitions["BUILD_UNIT_TESTS"] = False
+        if self.settings.compiler == "Visual Studio":
+            cmake.definitions["USE_MSVC_RUNTIME_LIBRARY_DLL"] = "MD" in self.settings.compiler.runtime
+        with tools.chdir(self._build_subfolder):
+            cmake.configure()
+        return cmake
+
+    def build(self):
+        tools.mkdir(self._build_subfolder)
+        cmake = CMake(self)
+        self._configure_cmake(cmake)
+        with tools.chdir(self._build_subfolder):
+            cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        self._configure_cmake(cmake)
+        with tools.chdir(self._build_subfolder):
+            cmake.install()
+        self.copy("LICENSE.txt", src=os.path.join(self.source_folder, self._source_subfolder), dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "bullet", "BulletConfig.cmake"))
+        self.copy("bullet_extra.cmake", src=self.source_folder, dst=os.path.join(self.package_folder, "lib", "cmake", "bullet"))
+
+    def package_info(self):
+        libs = []
+        if self.options.bullet3:
+            libs += [
+                "Bullet2FileLoader",
+                "Bullet3Collision",
+                "Bullet3Dynamics",
+                "Bullet3Geometry",
+                "Bullet3OpenCL_clew",
+            ]
+        libs += [
+            "BulletDynamics",
+            "BulletCollision",
+            "LinearMath",
+            "BulletSoftBody",
+            "Bullet3Common",
+            "BulletInverseDynamics",
+        ]
+        if self.settings.os == "Windows" and self.settings.build_type in ("Debug", "MinSizeRel", "RelWithDebugInfo"):
+            libs = [lib + "_{}".format(self.settings.build_type) for lib in libs]
+
+        self.cpp_info.name = "Bullet"
+        self.cpp_info.libs = libs
+        self.cpp_info.includedirs = ["include", os.path.join("include", "bullet")]
+        self.cpp_info.builddirs = [os.path.join("lib", "cmake", "bullet")]
+        self.cpp_info.build_modules = [os.path.join("lib", "cmake", "bullet", "bullet_extra.cmake")]
```

Comment:
>ls  /home/uilian/.conan/data/bullet3/2.88/_/_/package/50a5030bbbb13ae56bc4be41915ecd48549cb895/lib/cmake/bullet/
>BulletConfig.cmake  UseBullet.cmake

Where is bullet_extra.cmake? If there is no such file, then you should remove the entire cmake folder.

Reply:
There are some CMAKE variables defined by https://cmake.org/cmake/help/git-master/module/FindBullet.html and https://github.com/bulletphysics/bullet3/blob/6c722f8f94413b7096e6903e11bc087800511088/BulletConfig.cmake.in

When using `self.cpp_info.name = "Bullet"`, we need to define some variables manually.
`bullet_extra.cmake` is a file provided by this recipe.

Reply:
I see, thanks